Add docs for Middleman static site generator (#693)
ipfs/ipfs-docs: docs/how-to/websites-on-ipfs/static-site-generators.md

# Static-site generators

> Static-site generators like Hugo, VuePress, Nextjs, Middleman, and Jekyll are all incredibly popular platforms building websites quickly. This guide walks through how to integrate IFPS into each of these workflows.

Reviewer: ```suggestion
Static-site generators like Hugo, Jekyll, Middleman, Nextjs, and VuePress are all incredibly popular platforms building websites quickly. This guide walks through how to integrate IFPS into each of these workflows.
```

Upload the `dist` folder to IPFS.

## Middleman

Refer the Middleman's [Installation](https://middlemanapp.com/basics/install/) guide to install Ruby and Middleman.

1. Enable relative links in your project's `config.rb` file:

> ```ruby
> set :relative_links, true
> ```
> 
> Links generated by the `link_to` helper or by markdown will become relative except links to the root of your site.

Reviewer: ```suggestion
    ```ruby
    set :relative_links, true
    ```

    Links generated by the `link_to` helper or by markdown will become relative except links to the root of your site.
```

Upload the `dist` folder to IPFS.

## Middleman

Refer the Middleman's [Installation](https://middlemanapp.com/basics/install/) guide to install Ruby and Middleman.

> Refer the Middleman's [Starting a New Site](https://middlemanapp.com/basics/start-new-site/) guide to set up your project.

Reviewer: ```suggestion
```

> In `config.rb` enable relative links:

Reviewer: ```suggestion
1. Enable relative links in your project's `config.rb` file:
```

```ruby
set :relative_links, true
```

> Links generated by the `link_to` helper or by Markdown markup will become relative, except links to the root.

Reviewer: ```suggestion
Links generated by the `link_to` helper or by markdown will become relative except links to the root of your site.
```

> ```ruby
> set :relative_links, true
> ```
> 
> Links generated by the `link_to` helper or by Markdown markup will become relative, except links to the root.

Reviewer: ```suggestion
    ```ruby
    set :relative_links, true
    ```

    Links generated by the `link_to` helper or by Markdown markup will become relative, except links to the root.
```

> In `config.rb`, to generate relative links to the root, add the following helper:

Reviewer: ```suggestion
1. In `config.rb`, to generate relative links to the root, add the following helper:
```

> ```ruby
> helpers do
>   def relative_root_path
>     "#{"../" * (current_page.url.count('/') - 1)}index.html"
>   end
> end
> ```
> 
> Then use it in templates without the `link_to` helper:
> 
> ```erb
> <a href="<%= relative_root_path %>">Home</a>
> ```

Reviewer: ```suggestion
    ```ruby
    helpers do
      def relative_root_path
        "#{"../" * (current_page.url.count('/') - 1)}index.html"
      end
    end
    ```

1. Then use it in templates without the `link_to` helper:

    ```erb
    <a href="<%= relative_root_path %>">Home</a>
    ```
```

> To build a static site:
> 
> ```bash
> middleman build
> ```
> 
> Output will be in `./build` by default.
> 
> Upload the `build` folder to IPFS.

Reviewer: ```suggestion
1. Build your static site:

    ```bash
    middleman build
    ```

    Middleman will output your site to the `./build` folder.

1. Upload the `build` folder to IPFS.
```
